File: 100-Days-Series/Day 14/SecondLargestNum.c
#include<stdio.h>
void bubbleSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                // Swap arr[j] and arr[j + 1]
                arr[j]=arr[j]+arr[i+1];
                arr[i+1]=arr[j]-arr[i+1];
                arr[j]=arr[j]-arr[i+1];
            }
        }
    }
}
void main() {
    int n;
    // Input the size of the array
    printf("Enter the number of elements in the array: ");
    scanf("%d", &n);
    int arr[n];
    // Input array elements
    printf("Enter the elements of the array:\n");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    // print array elements
    for (int i = 0; i < n; i++) {
        printf("%d ,", arr[i]);
    }
    // Sort the array
    bubbleSort(arr, n);
    // Second largest number
    int size=sizeof(arr)/sizeof(arr[0]);
    printf("\nThe Second largest no. is %d",arr[size-2]);
}

File: 100-Days-Series/Day 14/SecondLargestNum.java
class SecondLargestNum{
    // Function for fine the seocnd largest number.
    public static int getSecondLargestNumber(int arr[]){
        // Bubble Sort
        for(int i=0;i<arr.length-1;i++){
            for(int j=0;j<arr.length-i-1;j++){
                if(arr[j]>arr[i+1]){
                    arr[j]=arr[j]+arr[i+1];
                    arr[i+1]=arr[j]-arr[i+1];
                    arr[j]=arr[j]-arr[i+1];
                }
            }
        }
        return arr[arr.length-1];
    }
// Main Method
    public static void main(String []args){
        java.util.Scanner s1=new java.util.Scanner(System.in);
        // Input the size of the array 
        System.out.print("Enter the number of elements in the array: ");
        int n=s1.nextInt();
        int arr[]=new int[n];
        // Input array elements
        System.out.print("Enter the elements of the array:\n");
        for(int i=0;i<n;i++){
            arr[i]=s1.nextInt();
        }
        // Print array element
        for(int i=0;i<arr.length;i++){
            System.out.print(" "+arr[i]);
        }
        int SecondLagestNumber=getSecondLargestNumber(arr);
        System.out.println("\nThe Seocnd Largest number is "+SecondLagestNumber);
    }
}